test/e2e/util/util.go

```diff
@@ -983,3 +986,43 @@ func RunQueryAtTime(ctx context.Context, log logr.Logger, prometheusClient prome
 		},
 	}, nil
 }
+
+func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx context.Context, hostClient crclient.Client, hcpNs string) {
+	t.Run("EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations", func(t *testing.T) {
+		g := NewWithT(t)
+		hcpPods := &corev1.PodList{}
+		if err := hostClient.List(ctx, hcpPods, &client.ListOptions{
```

Comment:
I don't want new pods with emptyDir to get the annotation transparently without us realising it and without this e2e warning us.
I want it to be audited so we can make a conscious acknowledgement. For that I'd expect this test to fail if it finds the annotation in pod which is not in a hardcoded list. So then a new pod has emptyDir and gets the annotation automatically this test will fail unless we add that new pod to the list explicitly.

Reply:
I've changed the approach to check volume by volume in the initial loop. 

- If the pod is not in the list and has the safe eviction - It will fail
- If the pod is in the list and the volume is not in the annotation - it will fail